client/blocks/app-banner/test/app-banner.jsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * @format
+ * @jest-environment jsdom
+ */
+
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { AppBanner } from 'blocks/app-banner';
+import { EDITOR, NOTES, READER, STATS } from 'blocks/app-banner/utils';
+
+describe( 'AppBanner', () => {
+	let wrapper;
+
+	beforeEach( () => {
+		wrapper = shallow( <AppBanner /> );
+	} );
+
+	test( 'iOS deep links return correct URIs for EDITOR', () => {
+		expect( wrapper.instance().getiOSDeepLink( '/post', EDITOR ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fpost'
+		);
+		expect( wrapper.instance().getiOSDeepLink( '/post/discover.wordpress.com', EDITOR ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fpost'
+		);
+		expect( wrapper.instance().getiOSDeepLink( null, EDITOR ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fpost'
+		);
+	} );
+
+	test( 'iOS deep links return correct URIs for NOTES', () => {
+		expect( wrapper.instance().getiOSDeepLink( '/notifications', NOTES ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fnotifications'
+		);
+		expect( wrapper.instance().getiOSDeepLink( '/notifications/12345', NOTES ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fnotifications'
+		);
+		expect( wrapper.instance().getiOSDeepLink( null, NOTES ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fnotifications'
+		);
+	} );
+
+	test( 'iOS deep links return correct URIs for READER', () => {
+		expect( wrapper.instance().getiOSDeepLink( '/', READER ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fread'
+		);
+		expect( wrapper.instance().getiOSDeepLink( '/read/feeds/12345/posts/6789', READER ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fread%2Ffeeds%2F12345%2Fposts%2F6789'
+		);
+		expect( wrapper.instance().getiOSDeepLink( null, READER ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fread'
+		);
+	} );
+
+	test( 'iOS deep links return correct URIs for STATS', () => {
+		expect( wrapper.instance().getiOSDeepLink( '/stats', STATS ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fstats'
+		);
+		expect( wrapper.instance().getiOSDeepLink( '/stats/day/discover.wordpress.com', STATS ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fstats%2Fday%2Fdiscover.wordpress.com'
+		);
+		expect( wrapper.instance().getiOSDeepLink( null, STATS ) ).toBe(
+			'https://apps.wordpress.com/get#%2Fstats'
+		);
+	} );
```

Comment:
Thanks for writing these tests. Don't let this be a blocker for the PR but I don't particularly find them that useful - they are mainly just making sure that somebody didn't change the constants defined in the code.

A  more valuable test might focus more on the behaviors - probably a good point to question the tests if the descriptions are like "should do the right thing"

So what are the behaviors here we want to test? What could unexpectedly go wrong?

```js
test( 'properly encodes tricky fragments', () => {
	expect( wrapper.instance().getiOSDeepLink( '?://&#', STATS ).split( '#' )[ 0 ] ).toEqual( '%3F%3A%2F%2F%26%23' );
} );
```

also, I know there was some discussion about methods vs. functions and I don't want to confuse you, but as you rightly questioned, testing is harder as methods. there's _no_ dependence on the instance here so let's move it out of the class and make it a function in the module scope that the class calls. the tests become more straight-forward

```js
import { getiOSDeepLink } from 'blocks/app-banner';

test( 'properly encodes tricky fragments', () => {
	expect( getiOSDeepLink( '?://&#', STATS ).split( '#' )[ 0 ] ).toEqual( '%3F%3A%2F%2F%26%23' );
} );
```